middleware/verify_privilege_token/claims.go

```diff
@@ -36,7 +36,7 @@ func getVehicleTokenClaims(c *fiber.Ctx) (VehicleTokenClaims, error) {
 
 	return VehicleTokenClaims{
 		VehicleTokenID: p.Sub,
-		UserEthAddress: p.UserID,
-		Privileges:     p.Privileges,
+		UserEthAddress: p.UserEthAddress,
+		PrivilegeIDs:   p.Privileges,
```

Comment:
It seems a bit odd that "id" is on one side but not the other. If you want to stick to leaving it out everywhere I think that's okay.